server/src/bots/telegram/services/storage.ts

```diff
@@ -8,24 +8,44 @@ import {
     listenSubscriptionsChanges,
     setQueryToAnalyse,
     setSubscription,
+    getAllUsers,
     getUser,
-    addUser
+    addUser,
 } from '../../../services/domain/storage';
 import * as firebase from 'firebase';
 
-export const getTelegramFullStorage: Function = getMessengerStorage(TELEGRAM_PREFIX);
-export const getTelegramSubscriptions: Function = getSubscriptions(TELEGRAM_PREFIX);
-export const getTelegramActiveSubscriptions: Function = getActiveSubscriptions(TELEGRAM_PREFIX);
-export const getTelegramUserSubscriptions: Function = getUserSubscription(TELEGRAM_PREFIX);
-export const getTelegramActiveUserSubscriptions: Function = getActiveUserSubscription(TELEGRAM_PREFIX);
-export const getTelegramUser: Function = getUser(TELEGRAM_PREFIX);
-export const addTelegramUser: Function = addUser(TELEGRAM_PREFIX);
-export const setTelegramSubscription: Function = setSubscription(TELEGRAM_PREFIX);
-export const setTelegramQueryToAnalyse: Function = setQueryToAnalyse(TELEGRAM_PREFIX);
+export const getTelegramFullStorage: Function = getMessengerStorage(
+    TELEGRAM_PREFIX
+);
+export const getTelegramSubscriptions: Function = getSubscriptions(
+    TELEGRAM_PREFIX
+);
+export const getTelegramActiveSubscriptions: Function = getActiveSubscriptions(
+    TELEGRAM_PREFIX
+);
+export const getTelegramUserSubscriptions: Function = getUserSubscription(
+    TELEGRAM_PREFIX
+);
+export const getTelegramActiveUserSubscriptions: Function = getActiveUserSubscription(
+    TELEGRAM_PREFIX
+);
+export const getTelegramUser = getUser(TELEGRAM_PREFIX);
```

Comment:
why some functions have types and some don't ?

Reply:
I feel like this layer is simply decorator for injecting platoform in our storage.
So I'd say this layer don't need to know anyhing about the types which pure starage has already cooked.

We could enhance those who has types here.

Reply:
Actually, I feel that it should and The storage should not know anything about them as **storage** retrieves only data from, from different parts of it. On top of that, these different parts may have different data structure (preferably don't, but maybe some of them need for some kind of structural things), thus, data types have to be here, and in storage, I deem that should be generic. What do you think?